### PR TITLE
Use setuptools_scm as git file finder

### LINKED ISSUE #23: Requires old version of setuptools-git
From setup.py:
```python
    install_requires=[
        'setuptools',
        'setuptools-git!=1.2',
    ],
```

However setuptools-git's latest version is 1.2 (https://pypi.python.org/pypi/setuptools-git). This kind of requirement make it impossible to create a Debian package for setuptools-odoo because Debian already contains version 1.2 of setuptools-git and it's not going to switch to an older version...

Instead of requiring an old version, you should:
- either work with setuptools-git to make a new version that is usable for you
- or you should embed the logic that you need in setuptools-odoo and get rid of the dependency altogether
- or find another way to achieve the desired effect

/cc @sbrun

### PR DESCRIPTION
**Pull request by sbidoul**
Remove dependency on unmaintained setuptools-git.

Closes #23

### CODE REVIEW

File: setuptools_odoo/git_file_finder.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    Spec here: http://setuptools.readthedocs.io/en/latest/setuptools.html#\
+        adding-support-for-revision-control-systems
+    """
+    toplevel = _git_toplevel(path)
+    if not toplevel:
+        return []
+    git_files, git_dirs = _git_ls_files_and_dirs(toplevel)
+    realpath = os.path.realpath(path)
+    assert realpath.startswith(toplevel)
```
Comment (lmignon) [inline setuptools_odoo/git_file_finder.py]: @sbidoul Is-it with knowledge that you use `assert`. These assertions will be stripped out and not evaluated if Python is started with the -O option.

**Comment by sbidoul** [comment]
@lmignon thanks for testing! I fixed a corner case in 5f97ff4.

Could you test again? Make sure you don't have setuptools-git nor setuptools-scm installed in the test environnement, since the file finders they provide could kick in in addition to this one.

**Comment by lmignon** [comment]
@sbidoul Ok pipeline relaunched.  In my requirements.txt file setuptools-git has been removed and I reference your branch of setuptools-odoo. But indeed, if one of these 2 packages is installed into server's python lib, my test is not significant.

**Comment by sbidoul** [comment]
Waiting for setuptools_scm 2.1 to be released.

**Comment by sbidoul** [comment]
Done manually at a36bbf21491af68b563c87d40cd3404af93caca9
